src/screens/Nearby/state.ts

```diff
@@ -404,7 +408,9 @@ export function useDepartureData(
     [location?.id, favoriteDepartures],
   );
 
-  useEffect(refresh, [location?.id]);
+  useEffect(refresh, [
+    location?.resultType === 'geolocation' ? null : location?.id,
+  ]);
```

Comment:
I not sure if this will work as intended. Remember that manually refreshing the departures list (in v1) or nearby stops list (in v2) should use the current position for the user.